TinyInsights/InsightsExtension.cs

```diff
@@ -28,4 +30,22 @@ public static MauiAppBuilder UseTinyInsights(this MauiAppBuilder appBuilder, str
         
         return appBuilder;
     }
+
+public static MauiAppBuilder UseTinyInsightsAsILogger(this MauiAppBuilder appBuilder, string applicationInsightsConnectionString, Action<IInsightsProvider>? configureProvider = null)
+    {
+
+        appBuilder.Services.AddSingleton<ILogger>((services) =>
+        {
+            var provider = new ApplicationInsightsProvider(applicationInsightsConnectionString);
+            
+            configureProvider?.Invoke(provider);
+
+            return provider;
+        });
+
+        appBuilder.Services.AddTransient<InsightsMessageHandler>();
```

Comment:
Should the message handler be really transient ?

Reply:
My idea is that the lifetime then will be handled by the HttpClient. But if you have opinions about that I will listen.